File: cubesort.java
import java.lang.reflect.Array;
import java.util.*;
//method to sort the cubes of the array elemrnt
public class cubesort{

    static void sortArray(int arr[], int n){
        Integer[] ar = new Integer[n];

        for(int i=0; i<n; i++){
            ar[i] = arr[i];
        }

        //actual method that will compare/sort
        Arrays.sort(ar, new Comparator<Integer>()
            {
                public int compare(Integer a, Integer b){
                    int x = (int)Math.pow(a,3);
                    int y = (int)Math.pow(b,3);
                    return (x<y) ? -1 : 1;
                }
            }
        );

        //print the array
        for(int i=0; i<n; i++) {
            System.out.println(ar[i]+ " ");
        }
    }
    //caller code
    public static void  main(String[] args){
        //the initial array is
        int arr[]={4,-2,0,5,-9,3};
        int n= arr.length;

        //call the sortArray method
        sortArray(arr,n);
    }
}

File: cubesort.cpp
#include<bits/stdc++.h>
using namespace std;
void cubesort(int arr[], int n)
{
    int i,j,temp;
    for(i=0;i<n;i++)
    {
        for(j=i+1;j<n;j++)
        {
            int a=pow(arr[i],3);
            int b=pow(arr[j],3);
            if(a>b)
            {
                temp=arr[i];
                arr[i]=arr[j];
                arr[j]=temp;
            }
        }
    }
}
int main()
{
    int arr[] = {4, -2, 0, 5, -9, 3};
    int n = sizeof(arr)/sizeof(arr[0]);
    cubesort(arr, n);
    for(int i=0;i<n;i++)
        cout<<arr[i]<<" ";
}